app/controllers/oauth2_controller.rb

```diff
@@ -5,19 +5,28 @@
 # @!visibility private
 class Oauth2Controller < ApplicationController
 
+  PASSWORD_PROVIDER_NAME = 'password'.freeze
+  FACEBOOK_PROVIDER_NAME = 'facebook'.freeze
+  GOOGLE_PROVIDER_NAME = 'google'.freeze
+  EDX_PROVIDER_NAME = 'edx'.freeze
+  DEFAULT_CLIENT_NAME = 'unspecified'.freeze
+
   force_ssl if: -> { RailsApiAuth.force_ssl }
 
   # rubocop:disable MethodLength
   def create
+    client = params[:accessing_application]
+    client ||= DEFAULT_CLIENT_NAME
```

Comment:
These two lines can just be `client = params[:accessing_application] || DEFAULT_CLIENT_NAME`